src/app/task/task-list-item/task-list-item.component.html

```diff
@@ -28,12 +28,19 @@
   <div class="task-list-item__body task-details">
 
     <div class="task-details__description-container">
-      <textarea class="task-details__description" readonly>Some txt</textarea>
+      <textarea #taskDescription class="task-details__description" readonly>Some txt</textarea>
     </div>
   </div>
 
   <div class="task-list-item__btn-wrapper btn-wrapper">
-    <button class="btn-secondary btn-secondary_purpose-edit">Edit</button>
+    <button class="btn-secondary btn-secondary_purpose-edit" (click)="onEditTask()">Edit</button>
     <button class="btn-secondary btn-secondary_purpose-delete">Delete</button>
   </div>
+    
+  <div *ngIf="editable">
+    <div class="btn-wrapper">
+      <button class="btn" (click)="undoChanges(task)">Cancel</button>
+      <button class="btn btn-save" (click)="updateTask(task, {name: taskName.value, description: taskDescription.value})">Save</button>
```

Comment:
the same: why do you need a task as argument here? 
{name: taskName.value, description: taskDescription.value} - not a good solution. what if I would like to pass there 10 parameters to change. It is better to check changes in component by using angular forms or ngModel directive.